docker-freeipa/bootstrap/classpath/freeipa-salt/salt/freeipa/freeipa-getkeytab-plugin/getkeytab.py

```diff
@@ -0,0 +1,184 @@
+# Copyright (C) 2019  Cloudera
+# see file 'COPYING' for use and warranty information
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import six
+
+from ipalib import Command, Flag
+from ipalib import errors, output
+from ipalib.parameters import Principal
+from ipalib.plugable import Registry
+from ipalib.text import _
+from ipaserver.plugins.service import normalize_principal, validate_realm
+
+if six.PY3:
+    unicode = str
+
+__doc__ = _("""
+Retrieves a Kerberos keytab.
+
+WARNING:  Retrieving the keytab resets the secret for the Kerberos principal.
+This renders all other keytabs for that principal invalid.  When multiple hosts
+or services need to share the same key (for instance in high availability or
+load balancing  clusters),  the  retrieve  option  must  be  used  to
+retrieve the existing key instead of generating a new one (please refer to the
+EXAMPLES section).
+
+NOTE: The retrieve option cannot be set before an initial keytab is created for
+the Kerberos principal.
+
+
+EXAMPLES:
+
+ Create and retreive a keytab
+   ipa get_keytab foo/example.com@EXAMPLE.COM
+
+ Retrieve an existing keytab
+   ipa get_keytab foo/example.com@EXAMPLE.COM --retrieve
+
+""")
+
+logger = logging.getLogger(__name__)
+
+register = Registry()
+
+
+@register()
+class get_keytab(Command):
```

Comment:
I figured out how to change the name of the API so that it doesn't have to match the class name. So I was able to have the name match the case. But I think you meant CapsCase not camelCase.